packages/cdktf/lib/terraform-stack.ts

```diff
@@ -119,24 +118,14 @@ export class TerraformStack extends Construct {
   }
 
   private findAll<T>({
-    byConstructor: ClassConstructor,
     byPredicate,
   }: {
-    byConstructor?: Constructor<T>;
-    byPredicate?: (node: unknown) => boolean;
+    byPredicate: (node: unknown) => boolean;
   }): T[] {
-    const predicate = ClassConstructor
-      ? (x: unknown) => x instanceof ClassConstructor
-      : byPredicate;
-
-    if (!predicate) {
-      throw new Error("Either byConstructor or byPredicate must be provided");
-    }
-
     const items: T[] = [];
 
     const visit = async (node: IConstruct) => {
-      if (predicate(node)) {
+      if (byPredicate(node)) {
```

Comment:
Why don't we just have the predicate be the only positional argument? It seems unnecessary to me in this case to have an object deconstruction